test/functional/legacy/032_insert_expansion_spec.lua

```diff
@@ -0,0 +1,90 @@
+-- Test for insert expansion
+-- :se cpt=.,w
+-- * add-expands (word from next line) from other window
+-- * add-expands (current buffer first)
+-- * Local expansion, ends in an empty line (unless it becomes a global expansion)
+-- * starts Local and switches to global add-expansion
+-- :se cpt=.,w,i
+-- * i-add-expands and switches to local
+-- * add-expands lines (it would end in an empty line if it didn't ignored it self)
+-- :se cpt=kXtestfile
+-- * checks k-expansion, and file expansion (use Xtest11 instead of test11,
+-- * because TEST11.OUT may match first on DOS)
+-- :se cpt=w
+-- * checks make_cyclic in other window
+-- :se cpt=u nohid
+-- * checks unloaded buffer expansion
+-- * checks adding mode abortion
+-- :se cpt=t,d
+-- * tag expansion, define add-expansion interrupted
+-- * t-expansion
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('Insert expansion', function()
+  setup(clear)
+
+  it('is working', function()
+    insert([[
+      start of testfile
+      run1
+      run2
+      end of testfile
+
+      test11	36Gepeto	/Tag/
+      asd	test11file	36G
+      Makefile	to	run]])
+
+    execute('se backspace=""')
+    execute('se cpt=.,w ff=unix | $-2,$w!Xtestfile | set ff&')
+    execute('se cot=')
+    feed('<C-W>nO#include "Xtestfile"<cr>')
+    feed('ru<C-N><C-N><C-X><C-N><esc><C-A><cr>')
```

Comment:
> Possibly the <esc><c-a> is being interpreted as "meta ctrl-a" chord. Need to break it up. Same for the other <esc> feedkeys below.

I don't think that's the problem, I've tried breaking them up but no luck. In addition, since I've tried to run those commands by hand in both Vim and nvim, there's no way it's a mistake with the code but the commands itself. Thus, I feel that the original test is incorrect, though this doesn't make any sense as well.

Some examples in which I think the original test is flawed:

1. Logically if we run `ru<c-n><c-n>` we should get `run2`. However, if you look at the expect or test32.ok you will see `run1 run3`. Removing the second `<c-n>` (i.e., `ru<c-n>`), however, will work as expected.
2. For `execute('se cpt=u nohid')
feed('<c-w>oOEN<c-n><cr>')
feed('unl<c-n><c-x><c-x><c-p><esc><cr>')`,
following the logic, none of the unloaded buffers will contain the text `ENDTEST` or `unless`, because during the previous write command it's on the wrong window, therefore the autocomplete will not work for these two lines. I can get around this by adding `feed('<c-w>j<cr>')` before the writes.

I have yet to find the flaw on the last 3 lines of the test. If I can find it then this test should be done. But am I allowed to add more and remove commands to the original test like this? And can the original test really be wrong?